Update BigQuery timeout + retry configs for v1.1 (#1289)
dbt-labs/docs.getdbt.com: website/docs/reference/warehouse-profiles/bigquery-profile.md

      priority: interactive
```

### Timeouts and Retries

<VersionBlock firstVersion="1.1">

> Every BigQuery query is made in two steps:

Reviewer: ```suggestion
BigQuery starts every query in two steps:
```

> 1. Submit the query job to BQ server and obtain a query job id. 

Reviewer: ```suggestion
1. Job creation: Submitting a query job to a BigQuery server and obtaining a query job ID. 
```

> 2. Wait for the query to execute and poll the query result. 

Reviewer: ```suggestion
2. Job execution: Waiting for the query to execute and polling the query result. 
```

> dbt supports configuring BigQuery query timeouts in each step, as well as the retry and deadline of the overall query. 

Reviewer: ```suggestion
In a BigQuery profile, you can use dbt to configure:
* Query timeouts for job creation or job execution.
* The number of times you want the query to retry and the deadline, or maximum time for the query to run, for the overall query. 
```

#### job_execution_timeout_seconds

> The timeout setting of step 2, `job_execution_timeout_seconds`, is the most common one.

Reviewer: @jtcohen6 do you mean this is the most common configuration to set or to exceed? Or something else?

Author: This language all originated from a community member's PR (the person who contributed the feature to `dbt-bigquery`), so I'm not sure exactly what he meant by this. My interpretation is, this is the configuration you're most likely to want to set, as an end user of dbt + BigQuery. I'll update language to reflect

:::info Renamed config

In older versions, this same config was just called `timeout_seconds`.

:::
  

> By default, the query execuion timeout is set to 300 seconds. If a dbt model takes longer than this timeout to complete, then BigQuery may cancel the query and issue the following error:

Reviewer: ```suggestion
By default, the query execution timeout is set to 300 seconds. If a dbt model takes longer than 300 seconds timeout to complete, then BigQuery might cancel the query and issue the following error:
```

```
 Operation did not complete within the designated timeout.
```
  

> To change this timeout, use the `job_execution_timeout_seconds` configuration:

Reviewer: ```suggestion
You can change the timeout seconds for the job execution step by configuring `job_execution_timeout_seconds` in the BigQuery profile:
```

```yaml
my-profile:
  target: dev
  outputs:
    dev:
      type: bigquery
      method: oauth
      project: abc-123
      dataset: my_dataset
      job_execution_timeout_seconds: 600 # 10 minutes
```

#### job_creation_timeout_seconds
  

> In addition, dbt also allow more advanced timeout control in step 1 via `job_creation_timeout_seconds`, in which dbt submits a query job to BQ JobInsert API server and receives a query job id in return. This step shall be very quick, usually under a few seconds. However, in some rare situation, it could take much longer to complete. 

Reviewer: ```suggestion
You can change the timeout seconds in the creation step by configuring  `job_creation_timeout_seconds`. The creation step is when dbt submits a query job to the BQ JobInsert API server and receives a query job ID in return. This step usually takes under a few seconds. However, in some rare situations, it could take longer to complete. 
```

#### job_retry_deadline_seconds

> Pass a maximum deadline into the BigQuery client's built-in retry capabilities. From [Google's docs](https://googleapis.dev/python/google-api-core/latest/retry.html):

Reviewer: ```suggestion
Set the `job_retry_deadline_seconds ` to pass a maximum deadline into the BigQuery client's built-in retry capabilities. From [Google's docs](https://googleapis.dev/python/google-api-core/latest/retry.html):
```

> Because a exponential sleep algorithm is used, the retry is limited by a deadline. The deadline is the maxmimum amount of time a method can block. This is used instead of total number of retries because it is difficult to ascertain the amount of time a function can block when using total number of retries and exponential backoff.

#### job_retries

> In addition to using the BigQuery client's built-in retry logic, dbt will retry queries itself, up to a certain number of times, if they result in unhandled BigQuery errors.

Reviewer: @jtcohen6 just flagging this as not sure if I got the logic right here:
```suggestion
In addition to using the BigQuery client's built-in retry logic, you can also configure dbt using `job_retries` to retry queries up to a certain number of times. The `job_retires` will be used when dbt detects unhandled BigQuery errors in query results.
```

Author: Turns out I didn't have the logic quite right here! That's what happens when I skim through the code + Google's docs...

Updating to reflect actual behavior (as far as I can understand it)
